src/OpenTelemetry.Instrumentation.SqlClient/Implementation/SqlClientDiagnosticListener.cs

```diff
@@ -148,24 +157,40 @@ public override void OnEventWritten(string name, object? payload)
             case SqlDataAfterExecuteCommand:
             case SqlMicrosoftAfterExecuteCommand:
                 {
+                    activity = activity ?? this.currentActivity.Value;
+
                     if (activity == null)
                     {
                         SqlClientInstrumentationEventSource.Log.NullActivity(name);
                         return;
                     }
 
-                    if (activity.Source != SqlActivitySourceHelper.ActivitySource)
+                    //if (activity.Source != SqlActivitySourceHelper.ActivitySource && this.currentActivity.Value == null)
+                    //{
+                    //    return;
+                    //}
+
+                    activity.Stop();
+
+                    if (SqlActivitySourceHelper.DbClientOperationDuration.Enabled)
                     {
-                        return;
+                        SqlActivitySourceHelper.DbClientOperationDuration.Record(activity.Duration.TotalSeconds);
```

Comment:
You're not currently adding any dimensions, but once you get to that point...

Assuming you plan on relying on the activity to retrieve the values, you may not have what you need here when both metrics and tracing is enabled and `activity.IsAllDataRequested = false`.